custom-config-server-client/src/main/java/com/microsoft/sample/customConfigServerClient/AccessTokenManager.java

```diff
@@ -6,35 +6,53 @@
 
 package com.microsoft.sample.customConfigServerClient;
 
-import com.microsoft.azure.AzureEnvironment;
-import com.microsoft.azure.credentials.ApplicationTokenCredentials;
 import java.io.IOException;
 import java.io.InputStream;
+import java.time.OffsetDateTime;
 import java.util.Properties;
 
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.identity.ClientSecretCredentialBuilder;
+
 public class AccessTokenManager {
 
     static {
         Properties prop = new Properties();
         InputStream in = AccessTokenManager.class.getResourceAsStream("/application.properties");
         try {
             prop.load(in);
-            tokenClientId = prop.getProperty("access.token.clientId");
-            String tenantId = prop.getProperty("access.token.tenantId");
-            String secret = prop.getProperty("access.token.secret");
-            String clientId = prop.getProperty("access.token.clientId");
-            credentials = new ApplicationTokenCredentials(
-                clientId, tenantId, secret, AzureEnvironment.AZURE);
+
+            clientId = prop.getProperty("access.token.clientId");
+            tenantId = prop.getProperty("access.token.tenantId");
+            secret = prop.getProperty("access.token.secret");
+

```

Comment:
Is there a way to access the Spring environment at this stage?

Reply:
No. `@value` cannot be used before config server client.

Reply:
Check this https://github.com/spring-projects/spring-boot/commit/35673b74724e0a2d7775dc1c637da1f7c16cac11#diff-9745939396bdf4e9827f448638fbf417120bf799a1c1611f575d9ae4a444c28a, I guess when can use the Binder to bind the properties.